server/utils/recipeUtils.js

```diff
@@ -46,8 +47,71 @@ const getByID = asyncHandler(async (recipeID, Recipe) => {
     return recipe
 })
 
+const getShoppingList = asyncHandler(async (userId, Recipe, Ingredient) => {
+	await Ingredient.sync()
+	Recipe.hasMany(Ingredient, {foreignKey: 'recipe_id'})
+	Ingredient.belongsTo(Recipe, {foreignKey: 'recipe_id'})
+	console.log(Ingredient);
+	const recipes = await Ingredient.findAll({
+		attributes: [
+			'ingredient_name', 
+			'ingredient_unit', 
+			[
+				Sequelize.literal('SUM(Recipe.portion*ingredient_amount)'), 
+				'total_amount'
+			]
+		], 
+		include: [
+			{
+				model: Recipe, 
+				required: true, 
+				attributes: [], 
+				where: {
+					user_id: userId, 
+					portion: {[Op.gt]: "0"}
+				}
+			}
+		], 
+		group: [
+			'ingredient_name', 
+			'ingredient_unit'
+		], 
+		raw: true
+	});
+	return recipes
+})
+
+const setPortion = asyncHandler(async (recipeID, portions, userId, Recipe) => {
+	await Recipe.sync()
+	const recipe = await Recipe.findOne({ where: {id: recipeID, user_id: userId}});
+	
+	if(recipe){
+
+		recipe.portion = portions
```

Comment:
I think we can check if the portion number is less than or equal to 0 here.

Reply:
Related to my other comment, but do we need to check this if we can determine whether or not a recipe is included in the shopping list through the `portion` value alone?

Reply:
Given the front-end can make sure `portions >= 0` and there is no other way this value can be less than 0, we won't need to check here.

Reply:
Oh wait were you talking about checking `portion` given as a parameter or the value `recipe.portion` returned from the DB query?

Reply:
The one given as a param that is received from req.body here: https://github.com/COMP4350-Landstreicher/PlatePlanner/pull/83/files/91ea55c03772e6a8c35e0f27a1320ec2a2b42bbb#diff-c97396803f2d2e34da9edecc53b28fdec34636874f3c652c9b915e3bec63008fR39